app/gui2/src/util/database/__tests__/reactiveDb.test.ts

```diff
@@ -38,14 +38,21 @@ test('Indexing is efficient', () => {
 
 test('Error reported when indexer implementation returns non-unique pairs', () => {
   const db = new ReactiveDb()
-  const consoleError = vi.spyOn(console, 'error')
-  const _invalidIndex = new ReactiveIndex(db, (_id, _entry) => [[1, 1]])
-  db.set(1, 1)
-  db.set(2, 2)
-  expect(consoleError).toHaveBeenCalledOnce()
-  expect(consoleError).toHaveBeenCalledWith(
-    'Attempt to repeatedly write the same key-value pair (1,1) to the index. Please check your indexer implementation.',
-  )
+  const actualConsoleError = console.error
+  console.error = () => {}
+  try {
+    const consoleError = vi.spyOn(console, 'error')
+    // Invalid index
+    new ReactiveIndex(db, (_id, _entry) => [[1, 1]])
+    db.set(1, 1)
+    db.set(2, 2)
+    expect(consoleError).toHaveBeenCalledOnce()
+    expect(consoleError).toHaveBeenCalledWith(
+      'Attempt to repeatedly write the same key-value pair (1,1) to the index. Please check your indexer implementation.',
+    )
+  } finally {
+    console.error = actualConsoleError
+  }
```

Comment:
Instead of this manual spy restore, it's better to enable automatic mock restoring before each test.

https://vitest.dev/config/#restoremocks